docs/testing-operators.md

```diff
@@ -1,120 +1,404 @@
+
+```
+./operator-courier/scripts/get-quay-token
+
+Username: johndoe
+Password: 
+{"token": "basic abcdefghijkl=="}
+```
+
+A token takes the following form and should be saved in an environment variable:
+
+```
+export QUAY_TOKEN="basic abcdefghijkl=="
+```
+### Linting
+
+`operator-courier` will verify the fields included in the Operator metadata (CSV). The fields can also be manually reviewed according to [the operator CSV documentation](https://github.com/operator-framework/community-operators/blob/master/docs/required-fields.md).
+
+The following command will run `operator-courier` against the bundle directory `my-operator/` from the example above.
+
+```
+operator-courier verify --ui_validate_io my-operator/
+```
+
+If there is no output, the bundle passed `operator-courier` validation. If there are errors, your bundle will not work. If there are warning we still encourage you to fix them before proceeding to the next step.
+
+### Push to quay.io
+
+The Operator metadata in it's bundle format will be uploaded into your namespace in [quay.io](http://quay.io).
+
+> The value for `PACKAGE_NAME` *must* be the same as in the operator's `*package.yaml` file and the operator bundle directory name. Assuming it is `my-operator`, this can be found by running `cat my-operator/*.package.yaml`.
 
-### Upstream operators
+> The `PACKAGE_VERSION` is entirely up for you to decide. Best practice is it coincides with your Operator version.
 
-Operators submitted to the `upstream-community-operators/` directory are tested against a [`minikube`][minikube] instance deployed on a [Travis CI][travis-ci] environment. The OLM is installed locally in this case.
+```
+OPERATOR_DIR=my-operator/
+QUAY_USERNAME=johndoe
+PACKAGE_NAME=my-operator
+PACKAGE_VERSION=1.0.0
+TOKEN=$QUAY_TOKEN
+
+operator-courier push $OPERATOR_DIR $QUAY_USERNAME $PACKAGE_NAME $PACKAGE_VERSION $TOKEN
+```
+
+Once that has completed, you should see it listed in your account's [Applications](https://quay.io/application/) tab.
+
+> If the application has a lock icon, click through to the application and its Settings tab and select to make the application public.
+
+Your Operator bundle is now ready for testing. To upload subsequent versions, bump semver string in the `PACKAGE_VERSION` variable, as `operator-marketplace` always downloads the newest bundle according to [semantic versioning](https://semver.org/).
+
+## Testing Operator Deployment on Kubernetes
+
+Please ensure you have fulfilled the [pre-requisites](#pre-requisites) before continuing with the instructions below.
+
+### 1. Get a Kubernetes cluster
+
+Start a Kubernetes `minikube` cluster:
+
+```
+minikube start
+```
+
+### 2. Install OLM
+
+Install OLM into the cluster in the `olm` namespace:
+
+```
+kubectl apply -f operator-lifecycle-manager/deploy/upstream/manifests/latest/
+```
 
-### OpenShift operators
+> You may need to run this command twice if error occur related to `unable to recognize [...] no matches for kind [...]`
```

Comment:
Isn't this a solvable problem? Maybe even just a script that applies them one at a time?

Reply:
It is, and we should solve it in the OLM repository once and for all. Providing an alternative solution meanwhile.